src/realm/util/file_mapper.cpp

```diff
@@ -109,8 +109,10 @@ struct mapping_and_addr {
 util::Mutex& mapping_mutex = *(new util::Mutex);
 std::vector<mapping_and_addr>& mappings_by_addr = *new std::vector<mapping_and_addr>;
 std::vector<mappings_for_file>& mappings_by_file = *new std::vector<mappings_for_file>;
-unsigned int file_reclaim_index = 0;
-std::atomic<size_t> num_decrypted_pages(0); // this is for statistical purposes
+static unsigned int file_reclaim_index = 0;
+static std::atomic<size_t> num_decrypted_pages(0); // this is for statistical purposes
```

Comment:
Should these 3 be a decrypted_memory_stats_t struct?

Reply:
considered it briefly, but then we'd commit to having atomics in the struct (which is passed around by value), or to only updating the full struct atomically, which I felt was over engineering.

Reply:
agree - let's not. I thought about the same just after posting :-)